documentation/api/io.strimzi.api.kafka.model.connect.build.Build.adoc

```diff
@@ -0,0 +1,191 @@
+
+Examples of valid image names are:
+
+* `docker.io/my-org/my-image/my-tag`
+* `quay.io/my-org/my-image/my-tag`
+* `image-registry.image-registry.svc:5000/myproject/kafka-connect-build:latest`
+
+Different Kafka Connect deployments should use a different image (for example at least with different tags).
+
+If the registry requires authentication, the field `pushSecret` can be used to set a name of the Secret with the registry credentials.
+The Secret should use the `kubernetes.io/dockerconfigjson` type and contain the `.dockerconfigjson` file with the Docker credentials.
+For more details see link:https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/#registry-secret-existing-credentials[Create a Secret based on existing Docker credentials^]
+
+[source,yaml,subs=attributes+,options="nowrap"]
+.Example `output` configuration
+----
+apiVersion: {KafkaConnectApiVersion}
+kind: KafkaConnect
+metadata:
+  name: my-connect-cluster
+spec:
+  #...
+  build:
+    output:
+      type: docker # <1>
+      image: my-registry.io/my-org/my-connect-cluster:latest # <2>
+      pushSecret: my-registry-credentials # <3>
+  #...
+----
+<1> The type of the output which should be used by Strimzi. (Required)
+<2> The full name of the image which will be used including the repository and tag. (Required)
+<3> Name of the secret with the container registry credentials. (Optional)
+
+.Using OpenShift ImageStream
+
+Alternatively, you can also use OpenShift ImageStream to store the new container image.
+The ImageStream has to be created manually before deploying the Kafka Connect.
+To use OpenShift ImageStream, you have to set the `type` to `imagestream`, and the `image` field should contain the name of the ImageStream and the tag which should be used.
+For example `my-connect-image-stream:latest`.
+
+[source,yaml,subs=attributes+,options="nowrap"]
+.Example `output` configuration
+----
+apiVersion: {KafkaConnectApiVersion}
+kind: KafkaConnect
+metadata:
+  name: my-connect-cluster
+spec:
+  #...
+  build:
+    output:
+      type: imagestream # <1>
+      imageStream: image-stream-name # <2>
+  #...
+----
+<1> The type of the output which should be used by Strimzi. (Required)
+<2> The name of the ImageStream and tag. (Required)
+
+=== `plugins`
+
+The connector plugins which will be added to the container image are configured in the `.spec.build.plugins` section of the `KafkaConnect` custom resource.
+Each connector plugin needs to have a name which is unique within the Kafka Connect deployment.
+Additionally, it contains a list of artifacts which it consists of.
+These artifacts will be downloaded by Strimzi, added to the new container image, and used in the Kafka Connect deployment.
+The connector plugin artifacts can also include additional components such as (de)serializers and so on.
+Each connector plugin will be downloaded into a separate directory so that the different connectors and their dependencies are properly sand-boxed.
```

Comment:
```suggestion
Each connector plugin is downloaded into a separate directory so that the different connectors and their dependencies are properly _sandboxed_.
```